### ISSUE TITLE
ReCommended extension is not compatible with ReSharper 2023.1

### ISSUE THREAD

**Issue by sergeyQx** [prodot/ReCommended-Extension#61]
https://youtrack.jetbrains.com/issue/RSRP-491292/2023.1-EAP4-No-Intellisense-After-Unknown-Activity#focus=Comments-27-6865401.0-0

```
11:32:13.501 |E| HighlightingSettingsManagerImpl| :1                 | Could not add a configurable severity EmptyRegion. An item with the same key has already been added.

--- EXCEPTION #1/2 [ArgumentException]
Message = “An item with the same key has already been added.”
ExceptionPath = Root.InnerException
ClassName = System.ArgumentException
Data.ThreadLocalDebugInfo = AsyncParallelLifecycleInitializationStrategy::PriorityMainScheduler::QueueTask)
Data.Type = ReCommendedExtension.Analyzers.Region.EmptyRegionSuggestion
Data.Attr.RegisterConfigurableSeverityAttribute.ID = EmptyRegion
Data.Attr.RegisterConfigurableSeverityAttribute.CompoundItemName = <null>
Data.Attr.RegisterConfigurableSeverityAttribute.Group = DeclarationRedundancy
Data.Attr.RegisterConfigurableSeverityAttribute.Title = “Region is empty (ReCommended Extension)”
Data.Attr.RegisterConfigurableSeverityAttribute.DefaultSeverity = 3
Data.ConfigurableSeverityItem.Id = EmptyRegion
HResult = E_INVALIDARG=COR_E_ARGUMENT=WIN32_ERROR_INVALID_PARAMETER=80070057
Source = mscorlib
StackTraceString = “
  at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource) in ThrowArgumentException.il:line IL_0010 mvid 861E
     at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add) in Insert.il:line IL_0000 mvid 861E
     at JetBrains.ReSharper.Feature.Services.Daemon.HighlightingSettingsManagerState.LazyStaticConfigurableSeverityItems(PartCatalog catalog, IEnumerable`1 customConfigurableSeverityItemProviders, ILogger logger) in LazyStaticConfigurableSeverityItems.il:line IL_006F mvid 4F1A
”

--- Outer ---

--- EXCEPTION #2/2 [LoggerException]
Message = “Could not add a configurable severity EmptyRegion. An item with the same key has already been added.”
ExceptionPath = Root
ClassName = JetBrains.Util.LoggerException
Data.ManagedThreadName = <NULL>
InnerException = “Exception #1 at Root.InnerException”
HResult = COR_E_APPLICATION=80131600
StackTraceString = “
  at JetBrains.Util.ILoggerEx.LogException(ILogger this, LoggingLevel level, Exception exception, ExceptionOrigin exceptionOrigin, String comment) in LogException.il:line IL_0025 mvid CF22
     at JetBrains.Util.ILoggerEx.Error(ILogger this, Exception ex, String message) in Error.il:line IL_0015 mvid CF22 or Error.il:line IL_0025 mvid CF22
     at JetBrains.Util.ILoggerEx.Error(ILogger logger, Exception exception, JetLogErrorInterpolatedStringHandler& messageHandler) in Error.il:line IL_0015 mvid CF22 or Error.il:line IL_0025 mvid CF22
     at JetBrains.ReSharper.Feature.Services.Daemon.HighlightingSettingsManagerState.LazyStaticConfigurableSeverityItems(PartCatalog catalog, IEnumerable`1 customConfigurableSeverityItemProviders, ILogger logger) in LazyStaticConfigurableSeverityItems.il:line IL_006F mvid 4F1A
     at System.Lazy`1.CreateValue() in CreateValue.il:line IL_0060 mvid 861E
     at System.Lazy`1.LazyInitValue() in LazyInitValue.il:line IL_0080 mvid 861E
     at JetBrains.ReSharper.Feature.Services.Daemon.HighlightingSettingsManagerState.GetConfigurableSeverityItems() in GetConfigurableSeverityItems.il:line IL_0000 mvid 4F1A
     at JetBrains.ReSharper.Feature.Services.EditorConfig.EditorConfigHighlightingConverter..ctor(Lifetime lifetime, IEditorConfigSchemaCollector manager, HighlightingSettingsManager highlightingSettingsManager, SettingsSchema schema)
     at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
     at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture) in Invoke.il:line IL_00F4 mvid 861E
     at JetBrains.Application.Components.ConstructiorBindigns.Activate(ISingletonDescriptor descriptor, Object[] dependencies) in Activate.il:line IL_0031 mvid ABDE
     at JetBrains.Application.Components.SingletonDescriptor`1.CreateInstance(Object[] dependencies) in CreateInstance.il:line IL_005F mvid ABDE
     at JetBrains.Application.Components.AsyncDescriptorBase.ActivatingComponentState.CreateInstanceWithReadLock() in CreateInstanceWithReadLock.il:line IL_0040 mvid ABDE
     at JetBrains.Application.Components.AsyncDescriptorBase.ActivatingComponentState.TryGetSyncImpl() in TryGetSyncImpl.il:line IL_00BE mvid ABDE
     at JetBrains.Application.Components.AsyncDescriptorBase.StateWithPriorityCache`1.TryGetSync() in TryGetSync.il:line IL_0010 mvid ABDE
     at JetBrains.Application.Components.AsyncDescriptorBase.TryActivate(Object[] dependencyValues) in TryActivate.il:line IL_0023 mvid ABDE
     at JetBrains.Application.Components.AsyncDescriptorBase.TryGetOrCreateValue() in TryGetOrCreateValue.il:line IL_002B mvid ABDE
     at JetBrains.Application.Components.AsyncDescriptorBase.GetValue() in GetValue.il:line IL_000C mvid ABDE
     at JetBrains.Application.Components.InitializationStrategyBase.InitDescriptorSync(IComponentContainer container, IComponentDescriptor descriptor) in InitDescriptorSync.il:line IL_000D mvid ABDE
     at JetBrains.Application.Components.AsyncInitializationStrategyBase.<CreateDescriptorAsync>g__CreateValueDescriptorSafe|11_1(IComponentContainer container, IValueDescriptor descriptor, ILogger logger) in AsyncInitializationStrategyBase.il:line IL_0037 mvid ABDE
     at System.Threading.Tasks.Task.Execute() in Execute.il:line IL_0010 mvid 861E
     at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx) in RunInternal.il:line IL_0079 mvid 861E
     at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx) in Run.il:line IL_0009 mvid 861E or Run.il:line IL_0034 mvid 861E
     at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot) in ExecuteWithThreadLocal.il:line IL_00E1 mvid 861E
     at System.Threading.Tasks.Task.ExecuteEntry(Boolean bPreventDoubleExecution) in ExecuteEntry.il:line IL_0096 mvid 861E
     at JetBrains.Util.Concurrency.ExecutionContextEx.<>c__DisplayClass0_0.<Run>b__0(Object _) in ExecutionContextEx.il:line IL_0011 mvid CF22
     at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx) in RunInternal.il:line IL_0079 mvid 861E
     at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx) in Run.il:line IL_0009 mvid 861E or Run.il:line IL_0034 mvid 861E
     at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state) in Run.il:line IL_0009 mvid 861E or Run.il:line IL_0034 mvid 861E
     at JetBrains.Util.Concurrency.ExecutionContextEx.Run(ExecutionContext context, ContextCallback callback, Object state) in Run.il:line IL_0028 mvid CF22
     at JetBrains.Threading.ReentrancyGuard.Execute(String name, Action action) in Execute.il:line IL_00ED mvid CF22
     at JetBrains.Threading.ReentrancyGuard.ExecutePendingActions() in ExecutePendingActions.il:line IL_0065 mvid CF22
     at JetBrains.Util.Concurrency.ExecutionContextEx.<>c__DisplayClass0_0.<Run>b__0(Object _) in ExecutionContextEx.il:line IL_0011 mvid CF22
     at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx) in RunInternal.il:line IL_0079 mvid 861E
     at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx) in Run.il:line IL_0009 mvid 861E or Run.il:line IL_0034 mvid 861E
     at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state) in Run.il:line IL_0009 mvid 861E or Run.il:line IL_0034 mvid 861E
     at JetBrains.Util.Concurrency.ExecutionContextEx.Run(ExecutionContext context, ContextCallback callback, Object state) in Run.il:line IL_0028 mvid CF22
     at JetBrains.Threading.JetDispatcher.Closure.Execute() in Execute.il:line IL_00F9 mvid CF22
     at JetBrains.Threading.JetDispatcher.ProcessQueue(Int32 nMinBucket) in ProcessQueue.il:line IL_0114 mvid CF22
     at JetBrains.Threading.JetDispatcher.ExternalIdleProcessing.OnExternalDriverIdle() in OnExternalDriverIdle.il:line IL_001C mvid CF22
     at JetBrains.VsIntegration.Shell.VsDoIdleDispatcherSink.Microsoft.VisualStudio.OLE.Interop.IOleComponent.FDoIdle(UInt32 grfidlef)

```

**Comment by michael-damatov** [comment]
Thanks for reporting the issue.

The version 2023.1 EAP 4 is not supported by the ReCommended Extension for ReSharper.

**Comment by michael-damatov** [comment]
Released version 5.10.4 (supports ReSharper 2023.1)
